README.md

```diff
@@ -1,119 +1,111 @@
-
-```
-arr.forEach(callback[, thisArg])
-```
+## Getting started
 
-It says that `forEach` takes a callback as the first argument.  And it says the callback
-will receive 3 arguments when called:
-* currentValue
-* index
-* array
+1. **Fork & Clone** this repo.
+2. **Plan Ahead**: Research each iterator before you write any code.
+3. **Check Your Solution**: Write Driver Code, or run the included Tests (see below).
 
-It also mentions `thisArg` which **is not an argument to the callback**, rather it
-is an argument to `forEach` itself.  We can also see that `thisArg` is **optional**.
-That means it's not required.  For today's work, we will not use it.
+#### Before You Start Coding
 
-## Building our own
+For the following challenges it is essential that you understand the requirements to fully implement the built-in array method. See [MDN Array Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)
 
-Now we're going to build our own iterators.
+Before you start each problem, ask yourself questions such as:
 
-Remember to start small and add features later. It's easier to build incrementally than to try to do everything all at once.
+* What are our inputs?
+* What is our output?
+* What happens on each loop?
+* What does the callback function do?
+* What gets passed into our callback function? i.e. what arguments does it receive? (it's inputs)
+  * Where does it come from?
+  * How do we know what to name it?
 
-* Create a function `myEach` which implements `Array.prototype.forEach`
-* Create a function `myMap` which implements `Array.prototype.map`
-* Create a function `myReduce` which implements `Array.prototype.reduce`
+You should be able to answer most of these questions based on the documentation you just read or by experimenting in the browser developer tools.
 
-Note: for `myReduce` it is suggested that you start by coding a solution to solve it without `initialValue` first.  You can add initialValue later on.  See the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) for more details.
+## Challenges
+1. Create a function `myEach` which implements `Array.prototype.forEach`
+2. Create a function `myMap` which implements `Array.prototype.map`
+3. Create a function `myReduce` which implements `Array.prototype.reduce`
 
+> **Pro-Tip**: It's easier to build incrementally than to try to do everything all at once. Remember to start small and add features later.
 
-BONUS:
+**Bonuses**:
 
 * Create a function `myFilter` which implements `Array.prototype.filter`
 * Create a function `mySome` which implements `Array.prototype.some`
 * Create a function `myEvery` which implements `Array.prototype.every`
 
-### Before you start, review:
 
-Before you start each problem, ask yourself questions such as:
+# Check Your Solution
+## Checking Your Code: Using Driver Code
```

Comment:
I don't like the use of the term "driver code"  students will not know what that means.